### PR TITLE
feat: Make build, lint and test commands customizable

### PR DESCRIPTION
**Pull request by nullJaX**

### CODE REVIEW

File: cookiecutter.json [hunk 1]
```diff
@@ -29,5 +29,8 @@
     "pypi_credentials_id": "{% if cookiecutter.ci|lower == 'azure' %}testpypi-tomtom-dev{% else %}artifactory-pypi{% endif %}",
     "docker_registry_credentials_id": "{% if cookiecutter.ci|lower == 'azure' %}tomtom-docker-registry-bintray{% else %}artifactory-docker{% endif %}",
     "ci_ssh_agent_user": "{% if cookiecutter.ci|lower == 'jenkins' %}ssh_svc_ci{% else %}N/A{% endif %}",
-    "azure_build_definition_id": "{% if cookiecutter.ci|lower == 'azure' %}5{% else %}N/A{% endif %}"
+    "azure_build_definition_id": "{% if cookiecutter.ci|lower == 'azure' %}5{% else %}N/A{% endif %}",
+    "build_command": "python setup.py build",
```
Comment (maikelvdh) [inline cookiecutter.json]: Can we apply the same here: https://github.com/tomtom-international/azure-pipeline-templates/blob/master/steps/python/python.build.yml? If not yet then it makes sense to add `"{% if cookiecutter.ci|lower == 'jenkins' %}<value>{% else %}N/A{% endif %}` for now.
Same applies of course to `lintCommand` and `testCommand`.

Comment (martinm82) [reply cookiecutter.json]: As mentioned in the jsl PR (https://github.com/tomtom-international/jsl/pull/18), I think the best solution would be to use a thin layer on top (eg. Makefile), define an interface (with targets) and let the projects overwrite/define the content of such a target.

Comment (maikelvdh) [reply cookiecutter.json]: That could be done as second step and for now just here only make it available for Jenkins context and make Azure Pipelines not available.

File: cookiecutter.json [hunk 2]
```diff
@@ -28,6 +28,11 @@
     "pypi_repo": "{% if cookiecutter.ci|lower == 'azure' %}https://pypi.org/legacy/{% else %}https://artifactory.your_org.com/artifactory/api/pypi/pypi-local{% endif %}",
     "pypi_credentials_id": "{% if cookiecutter.ci|lower == 'azure' %}testpypi-tomtom-dev{% else %}artifactory-pypi{% endif %}",
     "docker_registry_credentials_id": "{% if cookiecutter.ci|lower == 'azure' %}tomtom-docker-registry-bintray{% else %}artifactory-docker{% endif %}",
+    "ci_ssh_agent_user": "{% if cookiecutter.ci|lower == 'jenkins' %}ssh_svc_ci{% else %}N/A{% endif %}",
```
Comment (maikelvdh) [inline cookiecutter.json]: This got removed with #59, so this is not correctly merged.

File: {{cookiecutter.project_name}}/Jenkinsfile [hunk 3]
```diff
@@ -7,4 +7,8 @@ pythonSetupPyPipeline \
   dockerRepo: "python", \
   pypiRepo: "{{ cookiecutter.pypi_repo }}", \
   pypiCredentialsId: "{{ cookiecutter.pypi_credentials_id }} ", \
+  sshAgentUser: "{{ cookiecutter.ci_ssh_agent_user }}", \
```
Comment (maikelvdh) [inline {{cookiecutter.project_name}}/Jenkinsfile]: This got removed with #59, so this is not correctly merged. Also I would group your new commands as last one.

File: {{cookiecutter.project_name}}/Jenkinsfile [hunk 4]
```diff
@@ -7,4 +7,8 @@ pythonSetupPyPipeline \
...
+  buildCommand: "{{ cookiecutter.build_command }}", \
+  lintCommand: "{{ cookiecutter.lint_command }}", \
+  testCommand: "{{ cookiecutter.test_command }}"
```
Comment (maikelvdh) [inline {{cookiecutter.project_name}}/Jenkinsfile]: Missing `, \` at the end of line.

File: cookiecutter.json [hunk 5]
```diff
@@ -28,6 +28,11 @@
...
+    "azure_build_definition_id": "{% if cookiecutter.ci|lower == 'azure' %}5{% else %}N/A{% endif %}",
+    "build_command": "python setup.py build",
+    "lintCommand": "python setup.py lint --lint-output-format parseable",
+    "testCommand": "python setup.py test --addopts '--cov-report xml:build/coverage.xml --cov-report term --cov-branch --junitxml=build/test_results.xml'"
```
Comment (maikelvdh) [inline cookiecutter.json]: Missing `,` at end of line.

File: cookiecutter.json [hunk 6]
```diff
@@ -28,6 +28,10 @@
     "pypi_repo": "{% if cookiecutter.ci|lower == 'azure' %}https://pypi.org/legacy/{% else %}https://artifactory.your_org.com/artifactory/api/pypi/pypi-local{% endif %}",
     "pypi_credentials_id": "{% if cookiecutter.ci|lower == 'azure' %}testpypi-tomtom-dev{% else %}artifactory-pypi{% endif %}",
     "docker_registry_credentials_id": "{% if cookiecutter.ci|lower == 'azure' %}tomtom-docker-registry-bintray{% else %}artifactory-docker{% endif %}",
+    "azure_build_definition_id": "{% if cookiecutter.ci|lower == 'azure' %}5{% else %}N/A{% endif %}",
```
Comment (maikelvdh) [inline cookiecutter.json]: This is now double defined here and on line 36.

**Comment by nullJaX** [comment]
All tested

**Pull request merged**